src/Driver/Mysqli/HostRequired.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Driver\Mysqli;
+
+/**
+ * @internal
```

Comment:
Also, note that I marked it as `@internal` (there's a plan to do the same for the rest of the named exceptions in newer versions). This way, we can rework them at any time w/o breaking other's code who should rely only on higher-level classes and interfaces.